tests/Aspire.Hosting.Tests/Dashboard/Playwright/PlaywrightFixture.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Model;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Playwright;
+using Xunit;
+
+namespace Aspire.Hosting.Tests.Dashboard;
+
+public class PlaywrightFixture : IAsyncLifetime
+{
+    public async Task InitializeAsync()
+    {
+        PlaywrightInstance = await Playwright.CreateAsync();
+        Browser = await PlaywrightInstance.Chromium.LaunchAsync(new() { Headless = false });
+    }
+
+    public IBrowser Browser { get; set; } = null!;
+    private IPlaywright PlaywrightInstance { get; set; } = null!;
+
+    public async Task DisposeAsync()
+    {
+        await Browser.DisposeAsync();
+        PlaywrightInstance.Dispose();
+    }
+
+    private sealed class MockDashboardClient : IDashboardClient
+    {
+        public bool IsEnabled => true;
+        public Task WhenConnected => Task.CompletedTask;
+        public string ApplicationName => "<marquee>An HTML title!</marquee>";
+        public ValueTask DisposeAsync() => ValueTask.CompletedTask;
+        public Task<ResourceCommandResponseViewModel> ExecuteResourceCommandAsync(string resourceName, string resourceType, CommandViewModel command, CancellationToken cancellationToken) => throw new NotImplementedException();
+        public IAsyncEnumerable<IReadOnlyList<ResourceLogLine>>? SubscribeConsoleLogs(string resourceName, CancellationToken cancellationToken) => throw new NotImplementedException();
+        public Task<ResourceViewModelSubscription> SubscribeResourcesAsync(CancellationToken cancellationToken) => throw new NotImplementedException();
+    }
+
+    public async Task<(string DashboardUrl, IPage Page)> SetupDashboardForPlaywrightAsync()
```

Comment:
Not a huge fan of tuples in a public surface. But it's test code so probably OK.